tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -0,0 +1,505 @@
+
+	_ = createFrrHubPod(hubPodWorkerName[1],
+		workerNodeList[1].Object.Name, hubConfigMap.Definition.Name, []string{}, hub1BRstaticIPAnnotation)
+
+	By("Creating configmap and Frr Master pod")
+
+	configMapStaticRoutes := defineConfigMapWithStaticRouteAndNetwork(hubIPv4ExternalAddresses,
+		removePrefixFromIPList(ipv4NodeAddrList))
+	masterConfigMap := createMasterConfigMap("master-configmap", configMapStaticRoutes)
+
+	masterStaticIPAnnotation := createStaticIPAnnotations(tsparams.ExternalMacVlanNADName, nadHub,
+		[]string{masterPodIPv4Prefix}, []string{""})
+
+	frrPod := createMasterPodTest("master-pod", "master-0", masterConfigMap.Definition.Name,
+		[]string{}, masterStaticIPAnnotation, portNum)
+
+	return frrPod
+}
+
+func createStaticIPAnnotations(internalNADName, externalNADName string, internalIPAddresses,
+	externalIPAddresses []string) []*types.NetworkSelectionElement {
+	ipAnnotation := pod.StaticIPAnnotation(internalNADName, internalIPAddresses)
+	ipAnnotation = append(ipAnnotation,
+		pod.StaticIPAnnotation(externalNADName, externalIPAddresses)...)
+
+	return ipAnnotation
+}
+
+func createHubConfigMap(name string) *configmap.Builder {
+	configMapData := DefineBaseConfig(tsparams.DaemonsFile, "", "")
+	hubConfigMap, err := configmap.NewBuilder(APIClient, name, tsparams.TestNamespaceName).
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create hub config map")
+
+	return hubConfigMap
+}
+
+func createMasterConfigMap(name, configMapStaticRoutes string) *configmap.Builder {
+	configMapData := DefineBaseConfig(tsparams.DaemonsFile, configMapStaticRoutes, "")
+	hubConfigMap, err := configmap.NewBuilder(APIClient, name, tsparams.TestNamespaceName).
+		WithData(configMapData).Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create hub config map")
+
+	return hubConfigMap
+}
+
+func createFrrHubPod(name, nodeName, configmapName string, defaultCMD []string,
+	secondaryNetConfig []*types.NetworkSelectionElement) *pod.Builder {
+	frrPod := pod.NewBuilder(APIClient, name, tsparams.TestNamespaceName, NetConfig.FrrImage).
+		DefineOnNode(nodeName).
+		WithTolerationToMaster().
+		WithSecondaryNetwork(secondaryNetConfig).
+		RedefineDefaultCMD(defaultCMD)
+
+	By("Creating FRR hub pod in the test namespace")
+
+	frrContainer := pod.NewContainerBuilder(
+		"frr2", NetConfig.CnfNetTestContainer, []string{"/bin/bash", "-c", "sleep INF"}).
+		WithSecurityCapabilities([]string{"NET_ADMIN", "NET_RAW", "SYS_ADMIN"}, true)
+
+	frrCtr, err := frrContainer.GetContainerCfg()
+	Expect(err).ToNot(HaveOccurred(), "Failed to get container configuration")
+	frrPod.WithAdditionalContainer(frrCtr).WithLocalVolume(configmapName, "/etc/frr")
+
+	frrPod, err = frrPod.WithPrivilegedFlag().CreateAndWaitUntilRunning(5 * time.Minute)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create FRR test pod")
+
+	return frrPod
+}
+
+func createMasterPodTest(name, nodeName, configmapName string, defaultCMD []string,
```

Comment:
very similar to the previos func. please try to combine them

Reply:
combined